File: 645-set-mismatch/set-mismatch.swift
class Solution {
    func findErrorNums(_ nums: [Int]) -> [Int] {
        var ans: [Int] = [Int]()
        var store: [Int:Int] = [Int: Int]()
        for number in nums{
            if let curr = store[number]{
                store[number] = curr+1
                ans.append(number)
            }else{
                store[number] = 1
            }
        }
        for number in 1..<nums.count+1{
            guard let curr = store[number] else{
                ans.append(number)
                return ans
            }
        }
        return []
    }
}

File: 645-set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        ans = []
        for index in range(len(nums)):
            if nums[abs(nums[index]) - 1] < 0:
                ans.append(abs(nums[index]))
            else:
                nums[abs(nums[index]) - 1] *= -1

        for index in range(len(nums)):
            if nums[index] > 0:
                ans.append(index + 1)

        return ans